### PR TITLE
Feature build enhancements

### PR DESCRIPTION
**Pull request by winkerVSbecks**
- Fixed the gulp sass task to not minify source-maps 
- Added [scss-linting](https://github.com/causes/scss-lint)
- Slightly Modified the linting rules. Disabled the following rules: 
  - Comment use only `//`
  - New line at the end of the file
  - Don't write trailing zeros for numeric values with a decimal point
  - Numeric values should not contain unnecessary fractional portions
- Updated the Sass to fix linting errors
- Added CSS stats with visualization. [cssstats](https://www.npmjs.com/package/cssstats) generates the stats. There is a really small angular app in the `/CSS` folder which visualizes the stats. This app is a fork of the [cssstats.com](http://cssstats.com) node app.
- Added documentation for linting and stats

@bertrandk @neilff @lustandfury @SethDavenport @claudiareinoza @abmohan @AbdellaToronto please review and merge. 

The linter actually checks for BEM syntax. This should help us implement the syntax easier. How do you guys feels about setting up the linter on Circle CI and reject PRs if they have any linting errors?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,19 @@ The Sassy Skeleton
 
 ## Development
 
-1. `npm install`
-2. `gulp dev`
-3. Go to [http://localhost:3000](http://localhost:3000)
\ No newline at end of file
+1. Install dependencies:
+    - `gem install sass`
```
Comment (SethDavenport) [inline README.md]: Is it worth streamlining this?  you could set a custom post install command in package.json such that npm install runs the gem and bower stuff too.

**Comment by SethDavenport** [comment]
I definitely support Circle-ci linting.

File: README.md (hunk 1 above)
Comment (winkerVSbecks) [inline README.md]: Good point. I'll make the change.

I usually have to do `sudo gem install ...` Would that still work with post install?

**Comment by winkerVSbecks** [comment]
I'm going to merge this and push a release to bower and npm. If you guys have any concerns/suggestions we should use github issues.

**Pull request merged**